tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/homescreen_settings.py

```diff
@@ -5,35 +5,65 @@
+    _get_more_homescreen_locator = (By.CSS_SELECTOR, '[data-l10n-id="get-more-home-screens"]')
+
+    _wallpaper_preview_locator = (By.CSS_SELECTOR, '.wallpaper-preview')
+    _wallpaper_pick_locator = (By.CLASS_NAME, 'wallpaper-button')
+    _pick_cancel_button_locator = (By.CSS_SELECTOR,
+                                             '#action-menu-list > button:nth-child(4)')
 
     @property
     def screen_element(self):
         return self.marionette.find_element(*self._page_locator)
 
-    def select_icon_layout(self, value):
+    @property
+    def change_screen_element(self):
+        return self.marionette.find_element(*self._change_homescreen_page_locator)
+
+    def open_change_home_screen(self):
         element = Wait(self.marionette).until(
-            expected.element_present(*self._icon_layout_locator))
+            expected.element_present(*self._change_homescreen_locator))
         Wait(self.marionette).until(expected.element_displayed(element))
         element.tap()
-        self.select(value)
+        Wait(self.marionette).until(expected.element_displayed(*self._change_homescreen_page_locator))
 
-    def open_select_icon_layout(self):
+    def open_get_more_home_screen(self):
         element = Wait(self.marionette).until(
-            expected.element_present(*self._icon_layout_locator))
+            expected.element_present(*self._get_more_homescreen_locator))
         Wait(self.marionette).until(expected.element_displayed(element))
         element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._change_homescreen_page_locator))
 
-    def close_select_icon_layout(self):
+    def cancel_get_more_home_screen(self):
         self.marionette.switch_to_frame()
-        self.marionette.find_element(*self._icon_layout_confirmation_button_locator).tap()
+        self.marionette.find_element(*self._pick_cancel_button_locator).tap()
         self.apps.switch_to_displayed_app()
         element = Wait(self.marionette).until(
-            expected.element_present(*self._icon_layout_locator))
+            expected.element_present(*self._change_homescreen_page_locator))
         Wait(self.marionette).until(expected.element_displayed(element))
+
+    @property
+    def wallpaper_preview_src(self):
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._wallpaper_preview_locator))
+        Wait(self.marionette).until(expected.element_displayed(element))
+        return element.get_attribute('src')
+
+    def pick_wallpaper(self):
+        self.marionette.find_element(*self._wallpaper_pick_locator).tap()
+        return Activities(self.marionette)
+
+    def cancel_pick_wallpaper(self):
+        self.marionette.switch_to_frame()
+        self.marionette.find_element(*self._pick_cancel_button_locator).tap()
+        self.apps.switch_to_displayed_app()
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._wallpaper_preview_locator))
```

Comment:
Same here, _wallpaper_preview_locator element might always be displayed when the pick wallpaper dialog is visible, so might want to wait on the _pick_cancel_button_locator to become not displayed.